backend/src/openarchiefbeheer/zaken/utils.py

```diff
@@ -177,14 +177,20 @@ def format_selectielijstklasse_choice(resultaat: Resultaat) -> DropDownChoice:
 
 
 @lru_cache
-def retrieve_selectielijstklasse_choices(process_type_url: str) -> list:
+def retrieve_selectielijstklasse_choices(
+    process_type_url: Optional[str] = None,
+) -> list:
     selectielijst_service = Service.objects.filter(api_type=APITypes.orc).first()
     if not selectielijst_service:
         return []
 
     client = build_client(selectielijst_service)
     with client:
-        response = client.get("resultaten", params={"procesType": process_type_url})
+        params = {}
+        if process_type_url:
+            params["procesType"] = process_type_url
+
+        response = client.get("resultaten", params=params)
```

Comment:
If you go for the changes suggested in the other comment, then I would update this to:
```python
@lru_cache
def retrieve_selectielijstklasse_choices(query_params: dict | None) -> list:
    selectielijst_service = Service.objects.filter(api_type=APITypes.orc).first()
    if not selectielijst_service:
        return []

    client = build_client(selectielijst_service)
    with client:
        response = client.get("resultaten", params=query_params)
        response.raise_for_status()
        data_iterator = pagination_helper(client, response.json())

    results = []
    for page in data_iterator:
        results += [
            format_selectielijstklasse_choice(result) for result in page["results"]
        ]

    return results
```